### PR TITLE
Robofac second mission

### PR DESCRIPTION
**Pull request by John-Candlebury**
#### Summary

```SUMMARY: Content "Second Hub 01 mission"```

#### Purpose of change

Makes the Hub 01 chain quest longer.

#### Describe the solution

The second Hub 01 mission involves finding a destroyed military convoy in order to recover data stored in a dead scientist Memory bionic. The player has to fight a decent number of military zombies, with at least one armored zombie among them.

#### Additional context

Todo:

- [x]  Mind Splicer iuse action.
- [x]  Proper mission mapgen.

### CODE REVIEW

**Comment by Xenomorph-III** [comment]
Is there a skill check for extracting the memory module?

**Comment by John-Candlebury** [comment]
Its going to be a long action, with first aid determining your success chances, similar to safe-cracking in a way.

File: data/json/items/generic.json [hunk 1]
```diff
@@ -376,6 +376,20 @@
     "volume": 0,
     "to_hit": -3
   },
+  {
+    "type": "GENERIC",
+    "id": "data_card",
+    "symbol": ",",
+    "color": "white",
+    "name": "data card",
+    "description": "Some type of advanced data storage device. Useful for storing very large amounts of information.",
```
Comment (Night-Pryanik) [inline data/json/items/generic.json]:
```suggestion
    "description": "Some type of advanced data storage device.  Useful for storing very large amounts of information.",
```

File: data/json/items/generic.json [hunk 2]
```diff
@@ -2728,6 +2742,19 @@
     "volume": 0,
     "to_hit": -3
   },
+  {
+    "type": "GENERIC",
+    "id": "robofac_intercom_2_corpse",
+    "symbol": "%",
+    "color": "white",
+    "name": "desiccated corpse",
+    "description": "A badly mangled and desiccated partial corpse. It seems whatever thing killed him did so with a single swipe of a gigantic claw.",
```
Comment (Night-Pryanik) [inline data/json/items/generic.json]:
```suggestion
    "description": "A badly mangled and desiccated partial corpse.  It seems whatever thing killed him did so with a single swipe of a gigantic claw.",
```

File: data/json/items/software.json [hunk 3]
```diff
@@ -70,5 +70,14 @@
     "name_plural": "train data",
     "description": "Logistical data on subterranean train routes and schedules.",
     "price": 200
+  },
+  {
+    "id": "mind_scan_robofac",
+    "type": "GENERIC",
+    "name": "neural data",
```
Comment (Night-Pryanik) [inline data/json/items/software.json]: Plural name required.

File: data/json/items/software.json [hunk 4]
```diff
@@ -70,5 +70,14 @@
...
+    "symbol": "#",
+    "container": "data_card",
+    "description": "Data stolen from a dead scientist memory banks.  Is the owner of these thoughts still hidden here, amidst the unreadable data; or are these just a collection of the precious moments of someone's life?\n\nWhatever the case, the idea of perpetually keeping a part of you within a metallic pill makes you uncomfortable.",
```
Comment (Night-Pryanik) [inline data/json/items/software.json]:
```suggestion
    "description": "Data stolen from a dead scientist memory banks.  Is the owner of these thoughts still hidden here, amidst the unreadable data, or are these just a collection of the precious moments of someone's life?\n\nWhatever the case, the idea of perpetually keeping a part of you within a metallic pill makes you feel uncomfortable.",
```

File: data/json/items/software.json [hunk 5]
```diff
@@ -70,5 +70,14 @@
...
+    "flags": [ "IRREMOVABLE", "NO_DROP" ]
```
Comment (Night-Pryanik) [inline data/json/items/software.json]: If I'm not mistaken, `IRREMOVABLE` is a flag only suitable for gun or tool mods.

File: data/json/items/tools.json [hunk 6]
```diff
@@ -339,6 +339,20 @@
...
+    "id": "mind_splicer",
+    "symbol": "o",
+    "color": "white",
+    "name": "mind splicer kit",
+    "description": "Surgical forceps, cables and a modified smartphone inside a small plastic pouch.  Assembled to steal the mind of some poor man, these are tools of the creepy high-tech sandman.",
+    "material": "plastic",
+    "price": 12500,
+    "use_action": "MIND_SPLICER",
+    "volume": 4,
+    "weight": 6,
```
Comment (Night-Pryanik) [inline data/json/items/tools.json]: Only 6 grams?

File: data/json/npcs/missiondef_robofac.json [hunk 7]
```diff
@@ -61,5 +62,66 @@
...
+        "place_monsters": [ { "monster": "GROUP_MIL_STRONG", "x": [ 1, 22 ], "y": [ 1, 22 ], "density": 2 } ]
+      }
+    },
+    "origins": [ "ORIGIN_SECONDARY" ],
+    "dialogue": {
+      "describe": "...",
+      "offer": "When the portal storms started, the Government issued an evacuation order for critical XEDRA personnel and sent convoys to retrieve them, with our head of AI research among the recalled.  We recently discovered that he died when the convoy transferring him was ambushed in the initial chaos, but his corpse and memory bionic might remain intact enough for us to extract valuable knowledge.  We want you to travel to the location, make a copy of his Bionic Memory Unit, and return it to us.",
+      "accepted": "Remember, do extraction /exactly/ as specified, otherwise the bionic will self-destruct.",
+      "rejected": "Yes, we recognize that our request is exceptional.  Return if you change your mind.",
+      "advice": " You do know what a memory unit looks like, right?  Mate grey, pill-sized, right in front of the corpus callosum. We suggest a forceps through the eye socket, shaking slightly, then slowly and carefully...",
```
Comment (Night-Pryanik) [inline data/json/npcs/missiondef_robofac.json]:
```suggestion
      "advice": " You do know what a memory unit looks like, right?  Matte gray, pill-sized, right in front of the corpus callosum.  We suggest a forceps through the eye socket, shaking slightly, then slowly and carefully...",
```

File: data/json/npcs/missiondef_robofac.json [hunk 8]
```diff
@@ -61,5 +62,66 @@
...
+      "inquire": "Do you have the scan?",
+      "success": "Uhhh... Well I guess this is acceptable. You have our thanks and payment.",
```
Comment (Night-Pryanik) [inline data/json/npcs/missiondef_robofac.json]:
```suggestion
      "success": "Uhhh...  Well, I guess this is acceptable.  You have our thanks and payment.",
```

File: data/json/player_activities.json [hunk 9]
```diff
@@ -600,6 +600,15 @@
...
     "no_resume": true
   },
+  {
+    "id": "ACT_MIND_SPLICER",
+    "type": "activity_type",
+    "activity_level": "NO_EXERCISE",
+    "stop_phrase": "Stop using the mind splicer?",
+    "suspendable": false,
+    "based_on": "speed",
+    "no_resume": true
```
Comment (Night-Pryanik) [inline data/json/player_activities.json]: Why? I think it's perfectly resumable.

File: src/iuse.cpp [hunk 10]
```diff
@@ -4462,6 +4462,37 @@ int iuse::blood_draw( player *p, item *it, bool, const tripoint & )
...
+            player_activity act( activity_id( "ACT_MIND_SPLICER" ), to_moves<int>( time ) );
+            act.targets.push_back( item_location( *p, &data_card ) );
+            p->assign_activity( act );
+            return it->type->charges_to_use();
+        } else {
+            add_msg( m_info, _( "Nevermind." ) );
+            return 0;
+        }
+    }
+    add_msg( m_info, _( "Theres nothing to use the %s on here." ), it->tname() );
```
Comment (Night-Pryanik) [inline src/iuse.cpp]:
```suggestion
    add_msg( m_info, _( "There's nothing to use the %s on here." ), it->tname() );
```

File: data/json/items/software.json (hunk 5 above)
Comment (John-Candlebury) [reply data/json/items/software.json]: Well in any case it was redundant here, so I removed it.

File: data/json/items/software.json (hunk 4 above)
Comment (John-Candlebury) [reply data/json/items/software.json]: Its intended but Im aware it breaks normal conventions.

File: data/json/items/generic.json [hunk 11]
```diff
@@ -2728,6 +2742,19 @@
     "volume": 0,
     "to_hit": -3
   },
+  {
+    "type": "GENERIC",
+    "id": "robofac_intercom_2_corpse",
```
Comment (mlangsdorf) [inline data/json/items/generic.json]: intercom_2_corpse implies this is the corpse of the second intercom.

**Pull request merged**